core/templates/dev/head/components/profile_link.html

```diff
@@ -0,0 +1,8 @@
+<script type="text/ng-template" id="components/profileLinkText">
+	<a ng-if="!(linkedName === 'admin') && !(linkedName === 'OppiaMigrationBot')" ng-href="/profile/<[linkedName]>"><[linkedName]></a>
```

Comment:
Nit: throughout, please use 2 spaces for HTML indentation, not tabs. (It should be possible to configure your text editor to use spaces.)

Good question about simplifying the 'if' clause. How about:

    ng-if="isUsernameLinkable(linkedName)"

and in the controller have something like:

    $scope.isUsernameLinkable = function(username) {
      return GLOBALS.SYSTEM_USERNAMES.indexOf(username) === -1;
    };

Then, add a variable to GLOBALS in base.html and pass it to the frontend from `core/controllers/base.py`, following the existing patterns. I think the list itself should be stored in `feconf.py` (short for 'frontend config', but it's basically where we keep our constants).